src/wfs.c

```diff
@@ -181,6 +181,139 @@ wfs_t * wfs_init(const char * file, char * err)
   return sim;
 }
 
+/**
+ * wfs_init2D:
+ * Initialize a wind field simulation from an input file.
+ */
+wfs_t * wfs_init2D(const char * file, char * err)
+{
+  int i, r, pos, rank;
+  dft_ptr_t N;
+  wfs_t * sim;
+
+  char * fbuf = read_file(file, '#');
+  if (fbuf == NULL) WFS_ERROR(err, "Input Error: Cannot open file");
+
+  char * wfs_buf = reg_match(fbuf, "WindFieldSimulation\\s*\\{.*\\}");
+  if (wfs_buf == NULL) WFS_ERROR(err, "Input Error: Cannot find valid WindFieldSimulation block");
+
+  char * box_buf = reg_match(wfs_buf, "(\\{|\\s)Box\\s*\\{.*\\}");
+  if (box_buf == NULL) WFS_ERROR(err, "Input Error: Cannot find valid Box block");
+
+  char * lx_buf = reg_match(box_buf, "(\\{|\\s)lx\\s*=\\s*\\+?[0-9]+(\\.[0-9]*)?(\\}|\\s)");
+  if (lx_buf == NULL) WFS_ERROR(err, "Input Error: Cannot find valid valid lx");
+
+  char * rx_buf = reg_match(box_buf, "(\\{|\\s)rx\\s*=\\s*[1-9]+[0-9]*(\\}|\\s)");
+  if (rx_buf == NULL) WFS_ERROR(err, "Input Error: Cannot find valid rx");
+
+  char * lz_buf = reg_match(box_buf, "(\\{|\\s)lz\\s*=\\s*\\+?[0-9]+(\\.[0-9]*)?(\\}|\\s)");
+  if (lz_buf == NULL) WFS_ERROR(err, "Input Error: Cannot find valid lz");
+
+  char * ly_buf = reg_match(box_buf, "(\\{|\\s)ly\\s*=\\s*\\+?[0-9]+(\\.[0-9]*)?(\\}|\\s)");
+  if (ly_buf == NULL) ly_buf = lz_buf; // in the 2D situation, ly refers to the interval over which to compute the y integral
```

Comment:
We can move this to `wfs_init()` and make `ry` default to 0 if not set so `ny=1` and the `herm` functions should work.